sql/driver/driver_test.go

```diff
@@ -202,6 +205,50 @@ func TestPlaceholders(t *testing.T) {
 	}
 }
 
+func TestSessionSettings(t *testing.T) {
+	defer leaktest.AfterTest(t)
+	s := server.StartTestServer(nil)
+	// Create a new Database called t.
+	db, err := sql.Open("cockroach", "https://root@"+s.ServingAddr()+"?certs=test_certs")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if _, err := db.Exec(`CREATE DATABASE t`); err != nil {
+		t.Fatal(err)
+	}
+	_ = db.Close()
+
+	// Open a new db client against the server with
+	// a fresh set of connections with the new settings.
+	// We create a database 't' above because you cannot
+	// "SET DATABASE = t" without the database existing.
+	if db, err = sql.Open("cockroach", "https://root@"+s.ServingAddr()+"?certs=test_certs"); err != nil {
+		t.Fatal(err)
+	}
+	defer cleanup(s, db)
+	driver.SetSessionSettings(db, driver.SessionSettings{DB: "t"})
+	var wq sync.WaitGroup
+	numRoutines := 10
+	wq.Add(numRoutines)
+	for i := 0; i < numRoutines; i++ {
+		go func(i int) {
+			defer wq.Done()
+			// Settings work!
+			if _, err := db.Exec(fmt.Sprintf(`CREATE TABLE kv%d (k CHAR PRIMARY KEY, v CHAR)`, i)); err != nil {
+				log.Fatal(err)
+				return
+			}
+			if _, err := db.Exec(fmt.Sprintf(`INSERT INTO kv%d VALUES ($1, $2)`, i), "a", "b"); err != nil {
+				log.Fatal(err)
+				return
+			}
+		}(i)
+	}
+	wq.Wait()
+	// Ugh! This is really awful.
+	driver.SetSessionSettings(db, driver.SessionSettings{})
```

Comment:
Is there a reason you would have to clear the settings here? Just curious because I think you should get rid of the `SetSessionSettings` interface which will resolve the ugliness.